utbot-framework/src/main/kotlin/org/utbot/framework/codegen/tree/CgSpringIntegrationTestClassConstructor.kt

```diff
@@ -1,21 +1,59 @@
 package org.utbot.framework.codegen.tree
 
-import org.utbot.framework.codegen.domain.builtin.autowiredClassId
+import org.utbot.common.tryLoadClass
 import org.utbot.framework.codegen.domain.context.CgContext
-import org.utbot.framework.codegen.domain.models.CgFieldDeclaration
-import org.utbot.framework.codegen.domain.models.CgMethodsCluster
-import org.utbot.framework.codegen.domain.models.SpringTestClassModel
-import org.utbot.framework.plugin.api.UtSpringContextModel
+import org.utbot.framework.codegen.domain.models.*
+import org.utbot.framework.plugin.api.ClassId
+import org.utbot.framework.plugin.api.util.SpringModelUtils.autoConfigureTestDbClassId
+import org.utbot.framework.plugin.api.util.SpringModelUtils.autowiredClassId
+import org.utbot.framework.plugin.api.util.SpringModelUtils.dirtiesContextClassId
+import org.utbot.framework.plugin.api.util.SpringModelUtils.dirtiesContextClassModeClassId
+import org.utbot.framework.plugin.api.util.SpringModelUtils.springBootTestClassId
+import org.utbot.framework.plugin.api.util.SpringModelUtils.transactionalClassId
+import org.utbot.framework.plugin.api.util.utContext
 
 class CgSpringIntegrationTestClassConstructor(context: CgContext) : CgAbstractSpringTestClassConstructor(context) {
+    override fun constructTestClass(testClassModel: SpringTestClassModel): CgClass {
+        return buildClass {
+            id = currentTestClass
+
+            body = constructTestClassBody(testClassModel)
+
+            with (currentTestClassContext) {
+                annotations += collectSpringSpecificAnnotations()
+                annotations += collectedTestClassAnnotations
+
+                superclass = testClassSuperclass
+                interfaces += collectedTestClassInterfaces
+            }
+        }
```

Comment:
Am I missing something, why isn't it implemented like this?
```suggestion
        currentTestClassContext.collectedTestClassAnnotations += collectSpringSpecificAnnotations()
        return super.constructTestClass()
```